Assets/MixedRealityToolkit/Utilities/StandardShader/MeshSmoother.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// Component which can be used to automatically generate smoothed normals on a mesh and pack 
+    /// those normals into a UV set. Smoothed normals can be used for a variety of effects including 
+    /// extruding disjoint meshes along a vertex normal. This behavior is designed to be used in conjunction 
+    /// with the MRTK/Standard shader which assumes smoothed normals are packed into the 3rd UV set.
+    /// </summary>
+    [RequireComponent(typeof(MeshFilter))]
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/README_MRTKStandardShader.html#mesh-outlines")]
+    public class MeshSmoother : MonoBehaviour
+    {
+        private const int smoothNormalUVChannel = 2;
+
+        [Tooltip("Should this component automatically smooth normals on awake?")]
+        [SerializeField]
+        private bool smoothNormalsOnAwake = false;
```

Comment:
should this have a getter / setter?

Reply:
That is a good question, since you can't modify the value in code before Awake is called I ommited them. Can you think of when we would need them for a property like this?